core/src/main/java/org/elasticsearch/search/aggregations/ParsedAggregation.java

```diff
@@ -36,21 +35,25 @@
 
     //TODO move CommonFields out of InternalAggregation
     protected static void declareCommonFields(ObjectParser<? extends ParsedAggregation, Void> objectParser) {
-        objectParser.declareObject((parsedAgg, metadata) -> parsedAgg.metadata.putAll(metadata),
+        objectParser.declareObject((parsedAgg, metadata) -> parsedAgg.metadata = metadata,
                 (parser, context) -> parser.map(), InternalAggregation.CommonFields.META);
     }
 
-    String name;
-    final Map<String, Object> metadata = new HashMap<>();
+    private String name;
+    protected Map<String, Object> metadata;
```

Comment:
why protected? subclasses can use the getter to retrieve it? but it only gets set while parsing? was this change required?